File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/java_solutions/2789-largest-element-in-an-array-after-merge-operations.java
class Solution {
    public long maxArrayValue(int[] nums) {
        long max = nums[nums.length - 1];
        long current = max;

        for (int i = nums.length - 2; i >= 0; i--) {
            if (nums[i] <= current) {
                current += nums[i];
            } else {
                current = nums[i];
            }
            max = Math.max(max, current);
        }

        return max;
    }
}

File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/erlang_solutions/2789-largest-element-in-an-array-after-merge-operations.erl
-module(solution).
-export([solve/1]).

solve([]) -> 0;
solve(Nums) ->
    ReversedNums = lists:reverse(Nums),
    [Head | Tail] = ReversedNums,
    solve_recursive(Tail, Head, Head).

solve_recursive([], _CurrentMergedValue, MaxSoFar) ->
    MaxSoFar;
solve_recursive([H | T], CurrentMergedValue, MaxSoFar) ->
    If H =< CurrentMergedValue ->
        NewCurrentMergedValue = H + CurrentMergedValue,
        NewMaxSoFar = max(MaxSoFar, NewCurrentMergedValue),
        solve_recursive(T, NewCurrentMergedValue, NewMaxSoFar);
    Else ->
        NewCurrentMergedValue = H,
        NewMaxSoFar = max(MaxSoFar, NewCurrentMergedValue),
        solve_recursive(T, NewCurrentMergedValue, NewMaxSoFar)
    End.